### DISCUSSION TITLE
useSearchParams changes return type from 13.2.4 to 13.5.6

### DISCUSSION THREAD

**Discussion by SalmenBejaoui** [vercel/next.js#60872] [Help]
### Summary

I'm trying to update the next.js version from 13.2.5 to 13.5.6. Running the build script, I see this

<img width="963" alt="Screen Shot 2024-01-19 at 09 53 07" src="https://github.com/vercel/next.js/assets/30017004/649ec11c-ceeb-45fe-ace0-a007da0f2029">

 and `tsconfig.json` and `next-env.d.ts` had some changes 
 
<img width="1236" alt="image" src="https://github.com/vercel/next.js/assets/30017004/955866ef-a878-42c3-b280-c267d08db8dc">

Those changes break my build because `useSearchParams` might return null as a type

<img width="917" alt="Screen Shot 2024-01-19 at 10 23 56" src="https://github.com/vercel/next.js/assets/30017004/d18b9903-d121-44d4-84bc-ef800ea96fa7">

This happens only on my machine, the rest of the team and the CICD build correctly.

### Additional information
Node.js: 18.18.0

_No response_

**Comment by ezeparziale** [comment]
If it's happened only in your computer I would try reinstall all the deps

**Reply by SalmenBejaoui** [reply]
I did it already. 
Looking at the documentation https://nextjs.org/docs/app/api-reference/functions/use-search-params in **Good to know** section, it seems `useSearchParams` returns `ReadonlyURLSearchParams | null` type in `pages` folder. This return works only by adding `next/navigation-types` in `next-env.d.ts` that next.js doesn't require to be added.

**Reply by ezeparziale** [reply]
But in this case you have the same problem in de CICD

**Reply by SalmenBejaoui** [reply]
actually no. the pipeline builds correctly

**Comment by SalmenBejaoui** [comment]
I found out that I had an empty app folder causing the build to check types differently.
